File: MatplotlibCS/PlotItems/Scatter.cs
using Newtonsoft.Json;
using System.Collections.Generic;

namespace MatplotlibCS.PlotItems
{
    public class Scatter : Line3D
    {
        public Scatter(string name)
            : base(name)
        {
            MarkerSize = 5;
            Marker = Marker.Circle;
            LineStyle = LineStyle.Solid;
            LineWidth = new List<float> { 1 };
        }

        /// <summary>
        /// Point colors
        /// </summary>
        [JsonProperty(PropertyName = "color")]
        public new List<Color> Color { get; set; } = new List<Color> { PlotItems.Color.Black };

        /// <summary>
        /// Width of line
        /// </summary>
        [JsonProperty(PropertyName = "lineWidth")]
        public new List<float> LineWidth { get; set; } = new List<float> { 1 };

        [JsonProperty(PropertyName = "plotnonfinite")]
        public bool PlotNonFinite { get; set; } = false;
    }
}

File: MatplotlibCS/Python/scatter.py
import matplotlib.pyplot as plot

class Scatter:
    def __init__(self, jsonDict):
        self.__dict__ = jsonDict

    def plot(self, axes):
        linewidth = self.lineWidth[0] if len(self.lineWidth) == 1 else self.lineWidth
        colors = self.color[0]["value"] if len(self.color) == 1 else map(lambda x: x["value"], self.color)
        axes.scatter(
            self.x,
            self.y,
            self.z,
            c=colors,
            marker=self.marker,
            alpha=self.alpha,
            linewidths=linewidth,
            plotnonfinite=self.plotnonfinite
        )
